src/pybind/mgr/mgr_module.py

```diff
@@ -1564,7 +1564,7 @@ def get_metadata(self,
         :rtype: dict, or None if no metadata found
         """
         metadata = self._ceph_get_metadata(svc_type, svc_id)
-        if metadata is None:
+        if not metadata:
```

Comment:
The doc string says `None` is returned if no metadata is available. That should be changed? Or is this a bug in _ceph_get_metadata() that it doesn't return None for 'mds' service but does so for others when no metadata is available?

Reply:
The metadata (map) could be empty IMO.

Reply:
I think I misunderstood. Ignore my comment.

Reply:
So we want defaultdict(lambda: 'unknown') to be returned when _ceph_get_metadata() returns empty dictionary for mds service? okay